packages/replay/src/replay.ts

```diff
@@ -175,6 +179,19 @@ export class ReplayContainer implements ReplayContainerInterface {
       return;
     }
 
+    const useCompression = Boolean(this._options.useCompression);
+
+    // Flush any pending events that were previously unable to be sent
+    try {
+      const pendingEvent = await getPendingReplay({ useCompression });
+      if (pendingEvent) {
+        await this.sendReplayRequest(pendingEvent);
```

Comment:
This happens before we attach any listeners at all and will block until we make this request. Note this calls the function that does *not* retry, but maybe it should?